### DISCUSSION TITLE
BOTW Gyro stops working at shrines, CEMU is latest version, no CEMUHook available

### DISCUSSION THREAD

**Discussion by causeburn** [kmicki/SteamDeckGyroDSU#35] [Q&A]
This seems to be an issue for a few people.

I have emudeck installed, the latest Cemu version installed and SteamDeckGyroDSU installed.

The gyro functionality works great outside of shrines but when I need to do the puzzles, it stops working.

The instructions seem to be outdated and assume Cemuhook is still being created for the latest Cemu version. Since Cemuhook is depreciated how are people supposed to get this problem fixed?

I can see the old instructions here https://github.com/kmicki/SteamDeckGyroDSU but as I mentioned, Cemuhook doesnt appear to work with the latest Cemu.

**Comment by kmicki** [comment]
Hi,

Gyro working outside of shrines means that Steam is translating gyro into right joystick input and SteamDeckGyroDSU isn't used at all.

In Cemu 2 it needs to be configured using "DSU controller" in input settings. I think Emudeck does that automatically but I am not sure. I will update the instructions as soon as I have some time.

**Reply by causeburn** [reply]
Hey Kmicki, appreciate the reply, unfortunately DSU controller was already set in the input settings. Not sure why its not working, SteamDeckGyroDSU was installed through Emudeck, if that helps.

I might just downgrade Cemu to 1.27, I hope that doesnt create any other issues. 

Let me know if you're interested in troubleshooting, I wouldnt mind helping out.

**Reply by causeburn** [reply]
I want to add, you might be on to something with the "joystick input".

As I mentioned the DSU controller is already selected in the Cemu settings but there are many different Steam Deck controller layouts, I played with many of them, no luck, but I wanted to show you some of the layouts and options I am seeing, you might see something immediately that I dont.

**The Cemu controller settings:**
- https://i.ibb.co/W52QWCR/aaa.jpg


**When I go to the controller settings in Steam Deck, I see this (keep in mind, I have set gyro behaviour to mouse and joystick, neither seem to work.**
- https://i.ibb.co/CvPfr41/MVIMG-20221224-084410.jpg


**When I go to the gyro setting in the Steam Deck controller layouts, under "gyro behaviour", I see these options:**
- https://i.ibb.co/xXBDyV0/MVIMG-20221224-084204.jpg


**Here are some of the Valve created controller templates:**
- https://i.ibb.co/ggZzGJP/MVIMG-20221224-084516.jpg


**Here are some of the community created controller templates (note: the highlighted template in the photo is the one I currently have selected, I have tried Valve templates, other community templates, none worked with gyro):**
- https://i.ibb.co/D8yfRqN/MVIMG-20221224-084549.jpg
- https://i.ibb.co/XknMJVn/MVIMG-20221224-085552.jpg

**Reply by kmicki** [reply]
The controller settings in steam are completely independent from SteamDeckGyroDSU. They can even work together. But Steam can only use gyro to emulate joystick or mouse so it will not work in shrines.

Could you execute a command in Konsole in Desktop Mode when Cemu is running (game doesn't have to be started):

    systemctl --user status sdgyrodsu

and paste the output here?

**Reply by causeburn** [reply]
Well it doesnt appear to be running at least.

I launched Cemu for Emudeck and have the main program window open, not running any games (per your post).

This is the response I get for the command

- https://i.ibb.co/qscCvgm/MVIMG-20221224-120553.jpg

**Reply by kmicki** [reply]
That means it's not installed. I guess installation of sdgyrodsu by Emudeck is not working properly.

Try installing using the instructions on the readme page.

**Reply by Godsbane** [reply]
Hi Kmicki - we have fixed the installer. Sorry for any inconvenience. 

When we moved from a script installer to the AppImage, it stopped working because of some sudo -V / sudo -S / environment interaction we couldn't get around. 

Found a solution though, and it should work again.

**Reply by Philomorph** [reply]
I was pulling my hair out trying to find this solution - so much old info using cemuhook, etc, is still out there.
But even once I got here, Cemu couldn't find the controller from the server, even though I knew it was running. The key was systemctl command above - it showed me that the server was running on IP address 0.0.0.0, but Cemu by default was looking at the LOCALHOST address. Once I manually typed in the correct IP it found the controller and I was up and running.

Thanks!
